spec/unit/pool_manager_spec.rb

```diff
@@ -1503,6 +1503,393 @@
+
+    before(:each) do
+      allow(logger).to receive(:log)
+    end
+
+    it 'should set the pool template to match the configured template' do
+      subject.update_pool_template(poolconfig, provider, new_template, current_template)
+
+      expect(poolconfig['template']).to eq(new_template)
+    end
+
+    it 'should log that the template is updated' do
+      expect(logger).to receive(:log).with('s', "[*] [#{pool}] template updated from #{current_template} to #{new_template}")
+
+      subject.update_pool_template(poolconfig, provider, new_template, current_template)
+    end
+
+    it 'should run drain_pool' do
+      expect(subject).to receive(:drain_pool).with(pool)
+
+      subject.update_pool_template(poolconfig, provider, new_template, current_template)
+    end
+
+    it 'should log that a template is being prepared' do
+      expect(logger).to receive(:log).with('s', "[*] [#{pool}] preparing pool template for deployment")
+
+      subject.update_pool_template(poolconfig, provider, new_template, current_template)
+    end
+
+    it 'should run prepare_template' do
+      expect(subject).to receive(:prepare_template).with(poolconfig, provider)
+
+      subject.update_pool_template(poolconfig, provider, new_template, current_template)
+    end
+
+    it 'should log that the pool is ready for use' do
+      expect(logger).to receive(:log).with('s', "[*] [#{pool}] is ready for use")
+
+      subject.update_pool_template(poolconfig, provider, new_template, current_template)
+    end
+  end
+
+  describe 'remove_excess_vms' do
+    let(:config) {
+      YAML.load(<<-EOT
+---
+:pools:
+  - name: #{pool}
+    size: 2
+EOT
+      )
+    }
+
+    before(:each) do
+      expect(subject).not_to be_nil
+    end
+
+    context 'with a 0 total value' do
+      let(:ready) { 0 }
+      let(:total) { 0 }
+      it 'should return nil' do
+        expect(subject.remove_excess_vms(config[:pools][0], provider, ready, total)).to be_nil
+      end
+    end
+
+    context 'when the mutex is locked' do
+      let(:mutex) { Mutex.new }
+      let(:ready) { 2 }
+      let(:total) { 3 }
+      before(:each) do
+        mutex.lock
+        expect(subject).to receive(:pool_mutex).with(pool).and_return(mutex)
+      end
+
+      it 'should return nil' do
+        expect(subject.remove_excess_vms(config[:pools][0], provider, ready, total)).to be_nil
+      end
+    end
+
+    context 'with a total size less than the pool size' do
+      let(:ready) { 1 }
+      let(:total) { 2 }
+      it 'should return nil' do
+        expect(subject.remove_excess_vms(config[:pools][0], provider, ready, total)).to be_nil
+      end
+    end
+
+    context 'with a total size greater than the pool size' do
+      let(:ready) { 4 }
+      let(:total) { 4 }
+      it 'should remove excess ready vms' do
+        expect(subject).to receive(:move_vm_queue).exactly(2).times
+
+        subject.remove_excess_vms(config[:pools][0], provider, ready, total)
+      end
+
```

Comment:
can you also add my test here to check if it passes?
https://github.com/mattkirby/vmpooler/pull/2/files#diff-04094802b5bab73dcebfdd6e91f2b0a1R1673